src/index.ts

```diff
@@ -378,14 +378,19 @@ class ADCPlatform implements DynamicPlatformPlugin {
     const identities = await getIdentitiesState(authOpts.cookie, authOpts.ajaxKey);
     const identity = identities.data[0];
     if (identity) {
-      this.localizeTempUnitsToCelsius = identity.attributes.localizeTempUnitsToCelsius;
+      this.tempDisplayUnitSetting = identity.attributes.localizeTempUnitsToCelsius
+        ? hapCharacteristic.TemperatureDisplayUnits.CELSIUS
+        : hapCharacteristic.TemperatureDisplayUnits.FAHRENHEIT;
     }
   }
 
   /**
    * Method to update state on accessories/devices.
    */
   async refreshDevices(): Promise<void> {
+    // Get latest account settings, notably C/F display setting
+    await this.getAccountSettings();
```

Comment:
Realistically, how often do we think someone will change tir temperature setting? I'm trying to reduce the amount of calls we make to the API.

Reply:
Yeah I was trying to avoid that, but issue is if it _does_ change and don't have the right setting then thermostat is showing -7 or 150 degrees. Setting someones house to 150 doesn't seem desirable. Might be an option to not fetch on refresh, and add some sanity checking, and if any of the temps report extremes then we just disable thermostat support until restart or something. Or perhaps we do add a plugin user setting, add sanity check on startup to make sure it matches Alarm.com, but otherwise rely on plugin setting.

Reply:
Also just a note so I remember, if we aren't fetching repeatedly, we can get the setting from the loginSession response to avoid any extra calls.

Reply:
Ok, seems reasonable to merge this as-is for now. Eventually I'll want to do an audit of how often we call the API and see if we can reduce some of these, but this is a good trade off so people don't boil their homes!